src/confirmations_impl.h

```diff
@@ -7,218 +7,116 @@
-
-  void Step1StoreTheServersConfirmationsPublicKeyAndGenerator(
-      std::string confirmations_GH_pair,
-      std::vector<std::string> bat_names,
-      std::vector<std::string> bat_keys);
-
-  void Step2RefillConfirmationsIfNecessary(
-      std::string real_wallet_address,
-      std::string real_wallet_address_secret_key);
-  void Step2bRefillConfirmationsIfNecessary(
-      const std::string& url,
-      const int response_status_code,
-      const std::string& response,
-      const std::map<std::string, std::string>& headers);
-  void Step2cRefillConfirmationsIfNecessary(
-      const std::string& url,
-      const int response_status_code,
-      const std::string& response,
-      const std::map<std::string, std::string>& headers);
-  void OnStep2RefillConfirmationsIfNecessary(const Result result);
-
-  void Step3RedeemConfirmation(std::string real_creative_instance_id);
-  void Step3bRedeemConfirmation(
-      const std::string& url,
-      const int response_status_code,
-      const std::string& response,
-      const std::map<std::string, std::string>& headers);
-  void OnStep3RedeemConfirmation(const Result result);
-
-  void Step4RetrievePaymentIOUs();
-
-  void Step5CashInPaymentIOUs(std::string real_wallet_address);
-  void Step5bCashInPaymentIOUs(
-      const std::string& url,
-      const int response_status_code,
-      const std::string& response,
-      const std::map<std::string, std::string>& headers);
-  void OnStep5CashInPaymentIOUs(const Result result);
-
-  bool VerifyBatchDLEQProof(
-      std::string proof_string,
-      std::vector<std::string> blind_strings,
-      std::vector<std::string> signed_strings,
-      std::string public_key_string);
-
-  void PopFrontConfirmation();
-  void PopFrontPayment();
-
-  void SaveState();
+  std::unique_ptr<RefillTokens> refill_tokens_;
+
+  uint32_t retry_getting_signed_tokens_timer_id_;
+  void RetryGettingSignedTokens();
+  void StopRetryGettingSignedTokens();
+  bool IsRetryingToGetSignedTokens() const;
+  std::vector<UnblindedToken> unblinded_tokens_;
+
+  // Redeem token
+  std::unique_ptr<RedeemToken> redeem_token_;
+  std::vector<UnblindedToken> unblinded_payment_tokens_;
+
+  // Payout tokens
+  uint32_t payout_confirmations_timer_id_;
+  void PayoutConfirmations();
+  void StopPayingOutConfirmations();
+  bool IsPayingOutConfirmations() const;
+  std::unique_ptr<PayoutTokens> payout_tokens_;
+
+  // State
   void OnStateSaved(const Result result);
   void LoadState();
   void OnStateLoaded(const Result result, const std::string& json);
   void ResetState();
   void OnStateReset(const Result result);
 
   std::string ToJSON();
-  bool FromJSON(std::string json);
+  bool FromJSON(const std::string& json);
 
-  void VectorConcat(
-      std::vector<std::string>* dest,
-      std::vector<std::string>* source);
-  std::unique_ptr<base::ListValue> Munge(std::vector<std::string> v);
+  std::unique_ptr<base::ListValue> Munge(const std::vector<std::string>& v);
```

Comment:
Haha, does it need a more descriptive name? :-)

Reply:
This is one part of the code I did not refactor much but do agree at some point this should be improved